jni/src/main/java/com/horizen/sigproofnative/NaiveThresholdSigProof.java

```diff
@@ -229,20 +375,39 @@ public static boolean verifyProof(
             long ftMinAmount,
             FieldElement constant,
             long quality,
+            Optional<List<FieldElement>> customFields,
             byte[] proof,
             boolean checkProof,
             String verificationKeyPath,
             boolean checkVerificationKey
     )
     {
+        // Passing an empty array, Rust side, instead of an Option, will slightly simplify the process of unpacking.
+        FieldElement[] customFieldsArray = (customFields.isPresent()) ? customFields.get().toArray(new FieldElement[0]): new FieldElement[] {};
```

Comment:
Why not to simplify the interface of `verifyProof` and change `Optional<List<FieldElement>>` to `List<FieldElement>`?
Anyway inside if there are no custom fields we convert Optional it to empty list.

Reply:
Conversion to an empty list is performed because otherwise it would be a mess to unpack the Optional Rust side using all the JNI methods and so on. If you feel it's ok to pass an empty list when we don't have any custom field, ok I can modify the interface, no problem

Reply:
rust side case is clear and fine for me.
The point here was exactly about java method interface and this line.
So, let's change the interface only. Same in similar places.

Reply:
Will do, thanks !

Reply:
Done